### ISSUE TITLE
ActiveSupport::Duration incorrectly formats iso8601 representation for negative intervals

### ISSUE THREAD

**Issue by mlt** [rails/rails#39503]
ISO8601Serializer incorrectly encodes negative intervals making them incompatible with PostgreSQL database. https://github.com/rails/rails/blob/master/activesupport/lib/active_support/duration/iso8601_serializer.rb#L33

### Steps to reproduce

`-0.2.seconds.iso8601`

### Expected behavior

`PT-0.2S`

### Actual behavior

`-PT0.2S`

### System configuration
**Rails version**: 5.2.3

**Ruby version**: 2.6.4

**Comment by vipulnsward** [comment]
AFICAT this is expected: https://web.archive.org/web/20171019211402/https://www.loc.gov/standards/datetime/ISO_DIS%208601-1.pdf Individual parts are prefixed with the sign. 

Why is this "not expected" or fails for you?

**Comment by mlt** [comment]
That is not what PostgreSQL expects as an interval/duration http://sqlfiddle.com/#!17/9eecb/51333
```
ERROR:  invalid input syntax for type interval: "-PT0.2S" at character 8
STATEMENT:  select '-PT0.2S'::interval                                  
```

**Comment by vipulnsward** [comment]
Based on initial search, most libraries seem to support it:
https://github.com/moment/luxon/pull/683
https://github.com/moment/moment/issues/2408

The spec doesn't seem to have it:
https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm

PG is referring to https://www.ietf.org/rfc/rfc3339.txt from https://www.postgresql.org/docs/current/datatype-datetime.html

What I can understand from these is its better to have the sign "-" not as prefixed but on the digits when generating. But when consuming, to support both leading and after letters.

Given PG is not supporting this parsing(which is weird), its worth exploring defaulting to not having leading "-".

**Comment by mlt** [comment]
It seems unlikely PG will have support for that. [My patch was rejected](https://www.postgresql.org/message-id/1189720.1591238786%40sss.pgh.pa.us).

**Comment by pixeltrix** [comment]
@mit how is this affecting you on PostgreSQL? AFAICS we just call `to_s` and not `iso8601` when sending the value to the database.

**Comment by mlt** [comment]
@pixeltrix Good point! Thank you for reminding me. I completely forgot Rails (I use 5.2.x) has no Interval Oid built in. I've been using a variation on https://stackoverflow.com/q/43920743/673826 and https://github.com/crashtech/torque-postgresql/blob/master/lib/torque/postgresql/adapter/oid/interval.rb#L61 for so long with positive intervals to handy cast it back and forth that I forgot it was not part of Rails.

That said, it would be nice if Rails was more friendly towards Interval natively. I just tried with Rails 6.0.3.2 (on Windows), it still uses a (specialized) String. If I feed that directly to ActiveSupport::Duration.parse, it will fail expecting ISO8601. So there is still a need to set up default interval style for PG session etc. to make things transparent to Rails user. Perhaps back then, I decided it was a way to go passing ISO8601 back and forth until I came across negative intervals.

Definitely I can work my way around to use ISO8601 (luckily Rails understands PG's format) on parsing side only but not while casting to DB. I'll consider submitting a PR to [torque-postgresql](https://github.com/crashtech/torque-postgresql) gem.

**Comment by pixeltrix** [comment]
@mlt there's actually a PR to add proper interval support to Active Record in #16919 - perhaps add a comment there about negative intervals and we can get it pushed over the line.

**Comment by Envek** [comment]
It is so sad that https://github.com/rails/rails/pull/39538 with fix for that has been automatically closed.

**Comment by jeremy** [comment]
Fixed by #39538
